### PR TITLE
Feature/vault

### PR DESCRIPTION
**Pull request by ryan-s-roberts**
# PR Checklist

[x] Secret access trait
[x] Embedded secret backend that creates random keys in memory
[x] Vault backed secret backend
[x] Unifying type for secret backends / builders
[] Integration test against a vault backend
[] Replace existing secret access and CLI options

## Please complete this checklist after opening your PR

* [ ] I have updated documentation as necessary
* [ ] I have updated helm chart(s) if needed
* [ ] I have added tests if needed
* [ ] All new and existing tests are passing
* [ ] Any breaking changes are clearly flagged and documented
* [ ] I’ve included a link to any relevant ticket(s)

### CODE REVIEW

File: crates/chronicle-signing/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+use thiserror::Error;
+mod embedded_secret_manager_source;
+mod vault_secret_manager_source;
+
+static CHRONICLE_PK: &str = "chronicle-pk";
+static BATCHER_PK: &str = "batcher-pk";
+
+#[derive(Error, Debug)]
+pub enum SecretError {
+    #[error("Invalid file encoding {source}")]
```
Comment (mtbc) [inline crates/chronicle-signing/src/lib.rs:20]:
```suggestion
    #[error("Invalid file encoding: {source}")]
```
and others

File: Cargo.toml [hunk 2]
```diff
@@ -86,7 +88,8 @@ mime = "0.3"
 mockito = "1.1"
 oauth2 = "4.4"
 opa = { git = "https://github.com/tamasfe/opa-rs", rev = "3cf7fea" }
-openssl = "0.10.55"
+openssl = { version = "*" }
```
[the remark this answers is not in the document]
Comment (mtbc) [reply Cargo.toml:91]: I suppose that pre-commit or somesuch could look out for this.

[the remark this answers is not in the document]
Comment (ryanbtp) [reply Cargo.toml:91]: I'll fix that, it's a good way of finding the version that *will* work. Had a
hell of a time with the previous vault dependency

File: crates/chronicle/Cargo.toml [hunk 3]
```diff
@@ -7,47 +7,48 @@ version = "0.7.5"
...
-thiserror            = { workspace = true }
-tokio                = { workspace = true }
-toml                 = { workspace = true }
-tracing              = { workspace = true }
-tracing-log          = { workspace = true }
-url                  = { workspace = true, features = ["serde"] }
-user-error           = { workspace = true }
-uuid                 = { workspace = true }
-valico               = { workspace = true }
+Inflector           = { workspace = true }
```
Comment (suchapalaver) [inline crates/chronicle/Cargo.toml:10]: Shall we just start using default formatting so it's easier to see the actual changes?

File: crates/opactl/src/cli.rs [hunk 4]
```diff
@@ -12,11 +19,11 @@ use tracing::{debug, error, info};
...
 // Generate an ephemeral key if no key is provided
-fn transactor_key() -> Arg {
-    Arg::new( "transactor-key")
+fn batcher_key() -> Arg {
+    Arg::new( "batcher-key")
      .short('t')
-     .long("transactor-key")
-     .env("TRANSACTOR_KEY")
+     .long("batcher-key")
+     .env("batcher_KEY")
```
Comment (suchapalaver) [inline crates/opactl/src/cli.rs:26]:
```suggestion
     .env("BATCHER_KEY")
```

File: Cargo.toml [hunk 5]
```diff
@@ -27,17 +29,22 @@ async-graphql-poem = "5.0.9"
 async-stl-client = { git = "https://github.com/btpworks/async-stl-sdk" }
 async-stream = "0.3.3"
 async-trait = "0.1.61"
+backoff = { version = "0.4.0", features = ["futures", "tokio"] }
```
Comment (suchapalaver) [inline Cargo.toml:32]: I was struggling to figure out where you were using this as was interested in this crate.

File: crates/chronicle/Cargo.toml (hunk 3 above)
Comment (ryanbtp) [reply crates/chronicle/Cargo.toml:10]: Lets try to align it - I am using an extension currently - Even Better Toml,
there may be a CLI equivalent. It's annoying rustfmt doesn't

Comment (mtbc) [reply crates/chronicle/Cargo.toml:10]: CLI would be great but, for the meantime, we too could try the extension.

File: Cargo.toml [hunk 6]
```diff
@@ -142,4 +159,5 @@ url = "2.3.1"
 user-error = "1.2.8"
 uuid = "1.2.2"
 valico = "3.6.0"
+vaultrs = "*"
```
Comment (mtbc) [inline Cargo.toml:162]: A version number to fill in.

**Pull request merged**